### ISSUE TITLE
Doubt of error in serialize_to_binary() member function.

### ISSUE THREAD

**Issue by Jie-Geng** [tenable/routeros#19]
While reading the serialize_to_binary() function of the WinboxMessage.cpp, I doubt if some code is bug.

I think you likely miss the 'M2' prefix. In my opinion, these 2 lines of code should be:
`        std::string temp_string("M2"); `
`        temp_string.append(it->second.serialize_to_binary());`

I am not sure it is correct.

**Comment by jacob-baines** [comment]
M2 and the message size indicators are added in Winbox_Session::send().

**Comment by Jie-Geng** [comment]
Yes, you add M2 to the whole message. And when you serialize m_msgs, you added M2 too.
But when you serialize m_msg_array, you didn't add M2 for array elements.
I found that in the parse_binary, you remove M2 for elements.

**Comment by jacob-baines** [comment]
Oh, I see what you mean. Yeah, looks wrong. I'll work up a test case and push a patch tomorrow. Thanks @Jie-Geng!

**Comment by Jie-Geng** [comment]
Glad it helped you.
